service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/CreateAzureStorageContainerStep.java

```diff
@@ -0,0 +1,114 @@
+    var userRequest =
+        context
+            .getInputParameters()
+            .get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    var destinationWorkspaceId =
+        context
+            .getInputParameters()
+            .get(
+                WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+
+    var sourceDatabase =
+        FlightUtils.getRequired(
+            context.getInputParameters(),
+            WorkspaceFlightMapKeys.ResourceKeys.RESOURCE,
+            ControlledAzureDatabaseResource.class);
+
+    ControlledAzureStorageContainerResource controlledContainerResource =
+        new ControlledAzureStorageContainerResource(
+            WsmResourceFields.builder()
+                .workspaceUuid(destinationWorkspaceId)
+                .resourceId(resourceId)
+                .name(storageContainerName)
+                .cloningInstructions(CloningInstructions.COPY_NOTHING)
+                .properties(Map.of())
+                .createdByEmail(
+                    Objects.requireNonNull(
+                            context
+                                .getInputParameters()
+                                .get(
+                                    JobMapKeys.AUTH_USER_INFO.getKeyName(),
+                                    AuthenticatedUserRequest.class))
+                        .getEmail())
+                .state(WsmResourceState.CREATING)
+                .build(),
+            WsmControlledResourceFields.fromControlledResource(sourceDatabase),
+            storageContainerName);
+
+    ApiAzureStorageContainerCreationParameters destinationCreationParameters =
+        new ApiAzureStorageContainerCreationParameters().storageContainerName(storageContainerName);
+    ControlledResourceIamRole iamRole =
+        IamRoleUtils.getIamRoleForAccessScope(sourceDatabase.getAccessScope());
+
+    try {
+      ControlledResource destinationContainer =
+          controlledResourceService.createControlledResourceSync(
+              controlledContainerResource, iamRole, userRequest, destinationCreationParameters);
+
+      context
+          .getWorkingMap()
+          .put(
+              WorkspaceFlightMapKeys.ControlledResourceKeys.AZURE_STORAGE_CONTAINER,
+              destinationContainer);
+    } catch (DuplicateResourceException e) {
+      // We are catching DuplicateResourceException here since we check for the container's presence
+      // earlier in the parent flight of this step and bail out if it already exists.
+      // A duplicate resource being present in this context means we are in a retry and can move on
+      logger.info(
+          "Destination azure storage container already exists, resource_id = {}, name = {}",
+          resourceId,
+          storageContainerName);
+    }
+    return StepResult.getStepResultSuccess();
+  }
+
+  // Nothing to undo; can't undo a launch step
```

Comment:
looks like a leftover comment from when this was a launch step

Reply:
@dillydally414, I'm assuming this comment is removed in your PR?